Improve consistency of column names

vekien opened the pull request:
> PR created for discussion and continued work until it feels ready.

A majority of the `ex.json` file is very consistent and in excellent shape, this PR aims to meet a 100% shared param name for common names (eg: `ClassJobLevel` vs `Level`)

### Rules:

- Idealy, linked tables should be named the same unless it doesn't make sense to. eg:
  - Column: `MinionRace` links to `MinionRace` file. Makes sense being named the same
  - Column: `Behavior` links to `CompanionMove`. No one would say "what companion move type is that minion?" so the name doesn't make sense to be the same. `Behavior` is more appropriate.

### Recipes:

- `UnlockKey` > `SecretRecipeBook`, would this make more sense? Most columns that link to another field (eg Placename, or RecipeLevelTable etc) use the name of the table they link to vs a "custom name".

### Character Level / Class Job Level / "Level"

There can be areas of confusion when relating to the "Level", I think changing all params that do not read the "Level" CSV should be named more explicitly. Some examples:

- ClassJobLevel
- RetainerLevel
- ItemLevel

**Changes so far:**

- All "character" related levels changed to: `ClassJobLevel`

**Some to think about, I'm unsure on**

- `FishingSpot` and `FishParameter` both have a "Level" but unsure if this is class level? I don't do fishing!
- `GatheringItemLevelConvertTable` and `GatheringPointBase` both have a Level column.
- `MasterpieceSupplyDuty` has `MinLevel`, should this be `ClassJobLevel{Min}` ? (it also has a `MaxLevel` field)
- `SpearfishingNotebook` has a Level field, this a class level?

Commit:
Improve consistency of column names

UnlockKey > SecretRecipeBook
All "character" related levels named: ClassJobLevel, this would also match "ItemLevel" for "gear" related levels. This means those that are "Level" relate to the CSV
FishingSpot, FishParameter,  has a "Level" but unsure if this is class level? I don't do fishing
GatheringItemLevelConvertTable also has level, GatheringPointBase
MasterpieceSupplyDuty has MinLevel, is this "ClassJobLevel{Min}" ? (it also has a MaxLevel)
SpearfishingNotebook has level, this class level?

ufx commented:
Thanks, I agree with all these changes including SecretRecipeBook.

FishingSpot, GatheringPointBase, and SpearfishingNotebook Level are the display level of that specific gathering point.  Definitely not character levels.  FishParameter Level is some other level value and I'm not sure what its use is beyond (likely) spirit bonding.  Probably to calculate difficulty and bait as well.

MasterpieceSupplyDuty Min/MaxLevel are definitely ClassJobLevels.

Commit:
Rename MasterpieceSupplyDuty

Commit:
Update FishingSpot and GatheringPointBase to "NodeLevel"

Commit:
Update GatheringItemLevelConvertTable to "ItemLevel"

Commit:
Update SpearfishingNotebook to NodeLevel

vekien commented:
For Fishing/Gathinering we could use: `NodeLevel` as you have Gathering Nodes etc.

For `FishParameter` I was thinking `DifficultyLevel` ? Maybe, it goes up to 300+ and `ItemLevel` wouldn't really fit the meaning.

For `GatheringItemLevelConvertTable` is that level an `ItemLevel`?

Commit:
Update FishParameter to "DifficultyLevel"

vekien commented:
I'm just thinking of the rules of linked content, so some examples:

In **Companion** there is:
1) Column: Behavior links to CompanionMove
2) Column: MinionRace links to MinionRace

It makes sense why 1 is not named CompanionMove because we (as dataminers) know "Companion" is a bit misleading and SE may fix this (as they've started adding Minion files), and no one refers to the Behavior as "MinionMove", as that isn't how it is in game. SO Behavior makes sense.

I wonder if it should be `MinionBehavior` to match `MinionRace`. The idea is that columns that are customised (eg do not match their linked file) always contain a Subject, eg:

- `FCRank` subject = `FC` - matches file
- `CompanyCraftType` subject = `Company` - matches file
- `AchievementKind` subject = `Achievement` - matches file
- `SecretRecipeBook` could be: `RecipeMasterBook` if we named it differently than the file

ufx commented:
How about `GatheringLevel` for nodes and fishing spots (not the items in them; the nodes themselves)?

`GatheringItemLevelConvertTable`: Those are close to item levels but there are some deviations.  It also covers `EventItem` rows that do not have item levels.  `GatheringItemLevel` is probably the most accurate umbrella for these.  Perhaps `FishParameter` levels should be `GatheringItemLevel` too.

Commit:
Suggestions to switch from "Node" to "Gathering"

Also: GatheringItemLevelConvertTable from "ItemLevel" to "GatheringItemLevel"
Also: DifficultyLevel to GatheringItemLevel
